test/test_cli_overrides.py

```diff
@@ -0,0 +1,334 @@
+        f.write("")
+    return inputfile
+
+
+@pytest.fixture(scope="module")
+def outputdir(tmpdir_factory):
+    return tmpdir_factory.mktemp("output")
+
+
+@pytest.fixture(scope="module")
+def outputfile(outputdir):
+    return outputdir.join("outputfile.file")
+
+
+@pytest.fixture
+def test_data(inputdir, inputfile, outputdir, outputfile):
+    # Function, Method, Constant, FormattedString aren't included
+    # Field is not tested, should use mm.fields.Raw instead
+    data = {
+        "boolean": True,
+        "date": "1776-07-04",
+        "datetime": "1997-08-29T02:14:00-05:00",
+        "decimal": 2.7,
+        "dict": {"some_data": 5, "some_other_data": "hello"},
+        "email": "anemail@example.com",
+        "float": 3.14159,
+        "inputdir": str(inputdir),
+        "inputfile": str(inputfile),
+        "integer": 10,
+        "list": [300, 200, 800, 1000],
+        "localdatetime": "0001-01-01T00:00:00",
+        "log_level": "ERROR",
+        "nested": {"a": 1, "b": False},
+        "number": 5.5,
+        "numpyarray": [[1, 2], [3, 4]],
+        "outputdir": str(outputdir),
+        "outputfile": str(outputfile),
+        "raw": "{1,2,3}",
+        "slice": ":-1",
+        "string": "Four score and seven years ago",
+        "time": "05:00:00",
+        "timedelta": "945890400",
+        "url": "http://www.example.com",
+        "uuid": "f4a1c5ee-c214-4a2b-9e62-e4ba6beb771b"
+    }
+    return data
+
+
+class MyNestedSchema(DefaultSchema):
+    a = fields.Int(required=True)
+    b = fields.Boolean(required=True)
+
+
+class MySchema(ArgSchema):
+    boolean = fields.Boolean(required=True)
+    date = fields.Date(required=True)
+    datetime = fields.DateTime(required=True)
+    decimal = fields.Decimal(requied=True)
+    dict = fields.Dict(required=True)
+    email = fields.Email(required=True)
+    float = fields.Float(required=True)
+    inputdir = fields.InputDir(required=True)
+    inputfile = fields.InputFile(required=True)
+    integer = fields.Int(required=True)
+    list = fields.List(fields.Int, required=True)
+    localdatetime = fields.LocalDateTime(required=True)
+    nested = fields.Nested(MyNestedSchema, required=True)
+    number = fields.Number(required=True)
+    numpyarray = fields.NumpyArray(dtype="uint8", required=True)
+    outputdir = fields.OutputDir(required=True)
+    outputfile = fields.OutputFile(required=True)
+    raw = fields.Raw(required=True)
+    slice = fields.Slice(required=True)
+    string = fields.Str(required=True)
+    time = fields.Time(required=True)
+    timedelta = fields.TimeDelta(required=True)
+    url = fields.URL(required=True)
+    uuid = fields.UUID(required=True)
+
+
+def test_unexpected_input(test_data):
+    with pytest.raises(SystemExit):
+        ArgSchemaParser(test_data, schema_type=MySchema,
+                        args=["--notanarg", "something"])
+
+
+def test_override_boolean(test_data):
+    mod = ArgSchemaParser(test_data, schema_type=MySchema,
```

Comment:
should we add the other potentially valid things here.. true, 0, 1, True, false?

Reply:
Yes, probably. There's a test still that covers those in test_argschema_parser.py since for some reason I didn't move that one like I did the invalid input part.